floris/tools/optimization/scipy/yaw_wind_rose.py

```diff
@@ -225,30 +244,42 @@ def _get_initial_farm_power(self):
             else:
                 power_init = self.nturbs * [0.0]
 
-            self.initial_farm_powers.append(np.sum(power_init))
+            self.initial_farm_powers.append(
+                np.dot(self.turbine_weights, power_init)
+            )
 
-    def _get_power_for_yaw_angle_opt(self, yaw_angles):
+    def _get_power_for_yaw_angle_opt(self, yaw_angles_subset_norm):
         """
         Assign yaw angles to turbines, calculate wake, report power
 
         Args:
-            yaw_angles (np.array): Yaw to apply to each turbine.
+            yaw_angles_subset_norm (np.array): Yaw to apply to subset
+            of controlled turbines, normalized.
 
         Returns:
             power (float): Wind plant power. #TODO negative? in kW?
         """
-        yaw_angles = self._unnorm(
-            np.array(yaw_angles), self.minimum_yaw_angle, self.maximum_yaw_angle
+        yaw_angles_subset = self._unnorm(
+            np.array(yaw_angles_subset_norm),
+            self.minimum_yaw_angle,
+            self.maximum_yaw_angle
         )
 
-        power = -1 * self.fi.get_farm_power_for_yaw_angle(
-            yaw_angles,
+        # Create a full yaw angle array
+        yaw_angles = np.array(self.x0, dtype=float)
```

Comment:
Same thing here, where setting the yaw angles for the uncontrolled turbines to their baseline values makes more sense to me. Especially since only a single set of x0 values can be specified for all wind directions. Depending on the wind direction, different turbines will be downstream (assuming downstream turbines are excluded), so it seems safer to set those yaw angles to their baseline values.